### ISSUE TITLE
w-select menu in a w-dialog gets rendered below the dialog

### ISSUE THREAD

**Issue by nmauersberg** [antoniandre/wave-ui#57]
Hey there!
when I use a w-select in a w-dialog, the menu elements are rendered behind the dialog and were not accessible. I tried to fix it by adding:

```css
.w-select__menu {
  z-index: 100;
}
```
But I could not get it working. I will let you know, if I find some more details.
Cheers :)

Edit: Ah, went into your component and saw, that dialog has an index of 100. I was able to fix it temporary with a z-index of 1000, for w-select__menu

**Comment by nmauersberg** [comment]
Same issue when using a w-overlay

**Comment by antoniandre** [comment]
Hi @nmauersberg,
This is actually possible with the `detach-to` option. 
Example: https://codepen.io/antoniandre/pen/ZEexvEP?editors=1000

```vue
  <w-dialog class="my-dialog" content-class="pa6">
    <w-menu detach-to=".my-dialog">
      <template #activator="{ on }">
        <w-button
          v-on="on"
          outline
          color="primary">
          Show menu on click
        </w-button>
      </template>
      Menu content
    </w-menu>
  </w-dialog>
```

Alternatively, you could use the `z-index` option on the `w-menu` element to increase it.

The reason why it's not automatic, is that generating the `z-index` internally via JavaScript according to where it is mounted in (in a dialog or even a nested dialog) would actually be troublesome to override externally via CSS - and also heavier in JavaScript.
Hope it makes sense!

Side note: I just released version 1.32.4 where now, a click on the activator will hide the open menu (not with `show-on-hover`).

**Comment by nmauersberg** [comment]
Oh cool! Thank you for the explanation! :) Yes, makes perfectly sense. Hiding the menu on an additional click is also very nice! Already tried to customize that. The same functionality would be awesome for w-select elements! ;) Sometimes its hard to get out of them, when there is little space around.

Now after the update the labels do not work in the w-select menu, like before. Do I need to change it for w-select?

**Comment by nmauersberg** [comment]
Here is an Image of what happens: https://imgur.com/6lICHNI.png
Labels get undefined.

Cheers!

**Comment by nmauersberg** [comment]
From v. 1.32.4 on the labels in the "w-select" dropdown are showing `undefined` for me. Tried with the most basic w-select and it was still like that, so there might be a small bug? :)

https://imgur.com/6lICHNI.png

**Comment by nmauersberg** [comment]
Will close this issue now.
Thank you! :)

**Comment by antoniandre** [comment]
Hi @nmauersberg,

Thanks for reporting the other issues and feedback.
I've investigated what could have happened on your project and why you would see "undefined", and I could reproduce on a not-up-to-date Nuxt project, switching from Wave UI 1.32.3 to 1.32.4.
Which means the problem came from a change I made between these 2 versions. Looking at it, https://github.com/antoniandre/wave-ui/commits/master the only change I made is the update of the Vue dependency: https://github.com/antoniandre/wave-ui/commit/acb8059dde4dfd884b0037ab5b19fcde2d004e42

So to solve this issue, you need to either run `npm i vue vue-server-renderer vue-template-compiler` (the renderer & compiler must have the same version number as Vue), or delete your node_modules and package-lock.json (or yarn.lock) and run the install of dependencies again.

For the w-select click to close, I agree! I'll see what I can do.

**Comment by nmauersberg** [comment]
Hi @antoniandre :)
This solved it for me, I missed deleting the package-lock.json. Thank you for investigating!!

Wishing you an excellent week ahead.
Cheers

**Comment by antoniandre** [comment]
@nmauersberg, w-select is now also closable on click in version 1.32.7!
Have a great week too.

**Comment by nmauersberg** [comment]
@antoniandre Thank you! 😊 Awesome!!

**Comment by antoniandre** [comment]
Hey @nmauersberg,
I realized I showed you an example for `w-menu` and not `w-select`. I hope you figured in the end!
I have updated the missing description for `menu-props` to make it a little bit clearer (https://antoniandre.github.io/wave-ui/w-select#menu-props).

And just to be clear, the w-select component uses w-menu under the hood to attach the floating menu to a DOM node (by default to `.w-app`) and you could override that with `:menu-props="{ detachTo: '.my-container' }"`.

If in the future you find something not well explained in the docs don't hesitate to ask!

**Comment by nmauersberg** [comment]
@antoniandre Thank you friend! I appreciate your work and your help very much.

Yes, I figured it out, thanks :) Your components are structured very understandable, so I saw that w-menu is used in w-select.

You'll certainly hear from me anyways xD
Cheers

**Comment by nmauersberg** [comment]
@antoniandre One more question came up today, the "w-checkbox" seems to transform into several divs, with the first one having the "w-flex" and the "wrap" class attached. I am trying to remove the wrap class somehow, because I don't want it to wrap, when there is little space around. Do you know a way to achieve this?

And should I open an issue for small questions like this? Maybe some sort of Community / Chat / Forum would be cool, for wave-ui users to exchange and to help each other. I don't want to waste your time with small support requests and questions like this.

Kind regards,
Nico

**Comment by nmauersberg** [comment]
@antoniandre Nevermind! It was my own stupidity! :D

**Comment by antoniandre** [comment]
Sorry Nicolai, I've been very busy lately. 
I'll catch up later this week with a solution for you. :)

**Comment by antoniandre** [comment]
Hi @nmauersberg,

I've modified a few things to apply the `wrap` class (causes a flex-wrap) more precisely, on UI elements that are in a form.
That's the class that causes you trouble. :)

The purpose of the `wrap` class is to put on multiple lines an element when a label is present and is positioned outside of the field on the left or right, or to wrap the `w-checkboxes` and `w-radios` when `inline` and not fitting on 1 line.

In these screenshots you'll see how the new layout of all the components on different resolutions.

<img width="484" alt="Screen Shot 2021-06-27 at 5 44 25 PM" src="https://user-images.githubusercontent.com/22971289/123561099-94597180-d774-11eb-9ee4-21825621bd06.png">
<img width="479" alt="Screen Shot 2021-06-27 at 5 44 41 PM" src="https://user-images.githubusercontent.com/22971289/123561100-94597180-d774-11eb-9216-1b084c540811.png">
<img width="234" alt="Screen Shot 2021-06-27 at 5 46 26 PM" src="https://user-images.githubusercontent.com/22971289/123561101-94597180-d774-11eb-84d1-1ab0515ab3b3.png">

As you can see when a label is outside, it is best to display it on top of the field itself. But I also agree it does not make sense for single checkbox, single radio, and switch as you pointed. So I removed the wrap for them.

Let me know if there's anything else you see not working.

Cheers!

**Comment by nmauersberg** [comment]
@antoniandre Thank you :) This is really awesome!

The notification manager is a great improvement too. Unfortunately the update to 1.36.0 is not working for me. I am getting the following error:

```
 ERROR  [Vue warn]: Error in render: "TypeError: Cannot read property 'notifications' of null"                                                                                              15:13:09

found in

---> <WNotificationManager> at src/wave-ui/components/w-notification-manager.vue
       <WApp> at src/wave-ui/components/w-app.vue
         <Layouts/default.vue> at layouts/default.vue
           <Root>
```

After a look at the "w-notification-manager" component, it seems to me like the "this.notifManager" data prop is still initialized as `null`, when the computed method tries to access it. Even though it should be instantiated as a "new NotificationManager()" in `beforeMount`. Maybe some sort of (nuxt-related) timing issue?

I tried debugging, but changes to the source code in the node_module did apply, since the bundled version is used and I was not able to rebuild the module properly. I also tried deleting node_modules and the package-lock.json, but I keep getting this error, using nuxt js.

Cheers! :)
